### PR TITLE
Add simple mode to input generator

### PR DESCRIPTION
**Pull request by saphewilliam**
This PR adds a "simple" mode to the input generation that uses unchecked inputs to:

- Reduce the generated code file by ~70% (in my tested examples) 
- Greatly reduce the typescript computation load 
- Make it easier to extend the API with custom logic

All without losing much of the expressiveness of the GraphQL API.

Simple mode is enabled by setting config.inputs.simple to true, it is false by default since that would be a breaking change.

I'm testing this mode out on a personal project, I will report back to see if it is actually functioning as planned. In the meantime I'd love to hear your thoughts!

### CODE REVIEW

**Comment by saphewilliam** [comment]
After our conversation I will remove the exports I had added back in from this PR 👌🏼

**Comment by Cauen** [comment]
Hi @saphewilliam 
Tested and It seems to be great ✔
Inputs really got simpler and easier to use 🚀

>I will report back to see if it is actually functioning as planned.

Its ready for merge?

**Comment by saphewilliam** [comment]
I'm testing it this weekend / next week, then I'll be sure if it's ready 😄

**Comment by saphewilliam** [comment]
I've tested the simple mode on a project of mine and it seems to work perfectly now! Could you merge this PR and update the package in npm? 😄

**Comment by saphewilliam** [comment]
Would be nice to have some sort of auto-publishing setup using [Changesets](https://github.com/changesets/changesets) in a github action

**Comment by Cauen** [comment]
@saphewilliam 
It seems like using `PrismaUpdateOperationsInputFilter` is removing type safety from args
```ts
export const PostCreateInput = builder.inputRef<PrismaUpdateOperationsInputFilter<Prisma.PostCreateInput>>('PostCreateInput').implement({
  fields: PostCreateInputFields,
});

```

image

> Would be nice to have some sort of auto-publishing setup using [Changesets](https://github.com/changesets/changesets) in a github action

Thats great, i'll take a look

**Comment by saphewilliam** [comment]
@Cauen 

> What is the point of this generic?

## TL;DR
It matches the internal prisma type with the generated graphql endpoint.

## Long answer

The GraphQL schema generated from our code:
image

However, the Prisma schema also allows raw string inputs:
image

The generic strips away the raw string input to reflect the GraphQL schema better. 
Not doing this gave me some issues when trying to extend the generated code

image
Even though "set" is the only available option, I can't easily access it without doing a useless check if the arg is a string, it is in fact never a string.

This is that same code with the generic added:
image

## Your issue

I can't seem to reproduce your issue? It works fine for me
image
image

**Comment by saphewilliam** [comment]
Also I did it for all inputs because it's just easier and I thought it didn't have any effect on the other inputs.

**Comment by Cauen** [comment]
@saphewilliam Thanks for answer, looks very useful

About `args`, have you tried with the example in repo?
image
It shows all properties, But almost everything is "any"
image
Its seems like a version missmatch, what is you version configs?

**Comment by saphewilliam** [comment]
You're right, I think the prisma client is generating them only if necessary, so if the bool one is not used, it is not generated. Great catch! I'll fix this bug with a bit more sophisticated checking in the generic type.

**Comment by saphewilliam** [comment]
I made a new generic type that should fix this issue, could you test it on your setup just to be sure? 😄

**Comment by Cauen** [comment]
Awesome
